docs/root/major-version.md

```diff
@@ -0,0 +1,71 @@
+# Major Version Plan
+
+The current (as of October 2019) versions of all Google Cloud client
+libraries are incompatible with Grpc.Core.Api 2.x - which means
+they're incompatible with anyone trying to use the Microsoft gRPC
+libraries in .NET Core 3.0.
+
+This is effectively a forcing factor for us to make a series of
+breaking changes, and bump the major version of all libraries. While
+we *are* making forced breaking changes, we'll take the opportunity to make
+a few other improvements at the same time.
+
+The bulk of the changes will be within
+[GAX](https://github.com/googleapis/gax-dotnet) as that's where a
+lot of our common code for interacting with gRPC resides, but there
+will be some API-specific changes too.
+
+## Changes
+
+We plan on making the following changes:
+
+- Upgrade the Grpc.Core dependency to the latest 2.x release.
+- Remove the System.Interactive.Async dependency, replacing it with
+  Microsoft.Bcl.AsyncInterfaces.
+- Change our target platforms to .NET 4.6.1 and .NET Standard 2.0
+- Revisit our retry configuration to bring it in line with gRPC
+  "native" retry configuration.
+- Remove all obsolete members from all libraries.
+- Remove all retry-related members that are not generated under by new
+  microgenerator.
+- Where individual APIs have their own copies of common resource
+  names (such as LocationName), update them to use the common
+  resource names instead.
+
```

Comment:
Also: Removing method signature (flattened) overloads with nullable primitive type when a parameter is optional. These option primitive parameters are now just the primitive type.

Reply:
Done.